com.unity.render-pipelines.universal/Editor/Converter/Converters/RenderSettingsConverter.cs

```diff
@@ -0,0 +1,301 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEditor.Search;
+using UnityEngine;
+using UnityEngine.Rendering;
+using UnityEngine.Rendering.Universal;
+using ShadowQuality = UnityEngine.ShadowQuality;
+using ShadowResolution = UnityEngine.ShadowResolution;
+
+namespace UnityEditor.Rendering.Universal
+{
+    internal class RenderSettingsConverter : RenderPipelineConverter
+    {
+        public override string name => "Rendering Settings Upgrade";
+
+        public override string info =>
+            "This converter will look at creating Universal Render Pipeline assets and respective Renderer Assets and configure" +
+            " their settings based on equivalent settings from builtin renderer.";
+
+        public override Type conversion => typeof(BuiltInToURPConverterContainer);
+
+        // Used to store settings specific to Graphics Tiers
+        GraphicsTierSettings m_GraphicsTierSettings;
+
+        // Settings items, currently tracks Quality settings only
+        List<SettingsItem> m_SettingsItems = new List<SettingsItem>();
+
+        // List of the rendering modes required
+        List<RenderingMode> m_RenderingModes = new List<RenderingMode>();
```

Comment:
Isn't it safer to create the lists in `OnInitialize`? What if `OnInitialize` is called twice - we might still have entries in our list from the previous initialization.